litestar/handlers/base.py

```diff
@@ -251,6 +251,8 @@ def resolve_dependencies(self) -> dict[str, Provide]:
 
             for layer in self.ownership_layers:
                 for key, value in (layer.dependencies or {}).items():
+                    if not isinstance(value, Provide):
+                        value = Provide(value, sync_to_thread=True)  # noqa: PLW2901
```

Comment:
`sync_to_thread=True`? This will have a perf impact.

Reply:
Well, my thinking here was that this should "just work". Passing in a synchronous function which might block would result in blocking the main thread, which is worse than this performance degradation. It is a pitfall however, basically the opposite one of calling `Provide`. 

Perhaps we should handle `sync_to_thread` differently in general with `Provide`, and for example default to `False`, but give a warning if it's a sync function and it has not *explicitly* been set to `False`. The idea there would be that if a synchronous function is used that does not perform any blocking operation, it can easily just be replaced by an async function, but shooting yourself in the foot by accidentally missing a blocking operation is made harder due to the warning.

Wdyt?

Reply:
I think we should either require explicit Provide or default to false and warn in the docs that this is the behvaior.

Reply:
I really don't feel like warning in the docs is enough in relationship to the size of this footgun. What are your concerns with a runtime warning?

Reply:
i dunno, maybe its better - we simply dont have these in other places yet, but thats not a real reason tbh. So lets go with your suggestion.

Reply:
> we simply dont have these in other places yet, but thats not a real reason tbh

We should add more of those I think :upside_down_face:

Reply:
Implemented the warning functionality in #1648